File: Segundo-Bimestre/Parcial 1/Taller4_2B/Taller4-2B_python/Ejercicio5_5.py
print("")

# Declarar la matriz
M = []

print("PROCEDIMIENTO DETERMINANTE MATRIZ ORDEN 3")

for i in range (0,3):
    M.append([])    
    for j in range (0,3):
        valor = int(input("A["+str(i)+"]["+str(j)+"] = "))
        M[i].append(valor) 

# Desarrollar la función Sarrus
def Sarrus(M):
    A = []

    for i in range (0,3):
        A.append([]) 
        for j in range (0,3):
            valor = M [i][j]
            A[i].append(valor)

    print("***Matriz ampliada M***")
    for i in range (0,3):
        for j in range (0,3):
            print(M [i][j],end="   ")
        print(" ") 

    # Columnas adicionales de la matriz ampliada
    print("***MAtriz ampliada A***")
    for i in range (0,3):
        A.append([])
        for j in range (3,5):
            valor = A[i][j-3]
            A[i].append(valor)

    # Imprimir la matriz determinante
    for i in range (0,3):
        for j in range (0,5):
            print(A[i][j],end="   ")
        print(" ")

    # Cálculo del determinante
    i = 0
    k = 0
    j = 0
    dc = 0
    t = 3
    d = 1
    sumad = 0

    while (k<3):
        while ((j<t) and (t<=5)):
            d = d * A[i][j]
            print("D = ",d)
            i = i + 1
            j = j + 1

        sumad = sumad + d
        print("*** Suma D["+str(i)+"] = ",sumad)
        d = 1
        i = 0
        k = k + 1
        dc = dc + 1
        j = dc
        t = t + 1
    i = 0; j = 2; dc = 2; k = 0
    t = 0; z = 1; sumai = 0

    while (k<3):
        while ((j>=t) and (t<3)):
           z = z * A[i][j]
           i = i + 1
           j = j - 1
           print("z = ",z)
        sumai = sumai + z
        print("*** Suma I["+str(i)+"] = ",sumai)
        z = 1
        i = 0
        k = k + 1
        dc = dc + 1
        j = dc
        t = t + 1
    print( "sumad = ",sumad," - sumai = ",sumai)
    det = sumad - sumai
    return det
        
# Aplicarle un valor a la función
det = Sarrus(M)
print("El determinante de la matriz es igual a: ",det)       

File: Segundo-Bimestre/Parcial 1/Taller4_2B/Taller4_2B_java/src/taller4_2b/Ejercicio5_5.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package taller4_2b;
import java.util.Scanner;
/**
 *
 * @author DELL
 */
public class Ejercicio5_5 {
    public static void main(String[] args) {
        Scanner leer = new Scanner (System.in);
        
        // Declarar e inicializar las variables
        int i, j;
        double det;
        int [][] M = new int [3][3];
        
        System.out.println("PROCEDIMIENTO DETERMINANTE MATRIZ ORDEN 3");
        
        for (i=0; i<3; i++) {
            for (j=0; j<3; j++) {
                System.out.print("M ["+i+"] ["+j+"] = ");
                M[i][j] = leer.nextInt();
            }
        }
        det = Sarrus(M);
        System.out.println("El determinante de la matriz es igual a: "+det);
    }
    
    public static double Sarrus(int M[][]) {
        int i, j, t, dc, n, m;
        int [][] A = new int [3][5];
        double d, sumad, z, sumai, det;
        
        for (i=0; i<3; i++) {
            for (j=0; j<3; j++) {
                A[i][j] = M[i][j];
            }
        }
        
        System.out.println("***Matriz ampliada M***");
        for (i=0; i<3; i++) {
            for (j=0; j<3; j++) {
                System.out.print("  "+M[i][j]);
            }
            System.out.println("  ");
        }
        
        // Columnas adicionales de la matriz ampliada
        System.out.println("***Matriz ampliada A***");
        for (i=0; i<3; i++) {
            for (j=3; j<5; j++) {
                A[i][j] = A[i][j-3];
            }
        }
        
        // Imprimir la matriz determinante
        for (i=0; i<3; i++) {
            for (j=0; j<5; j++) {
                System.out.print("  "+A[i][j]);
            }
            System.out.println("  ");
        }       
       
        
        // Cálculo del determinante
        i = 0;
        int k = 0;
        j = 0; 
        dc = 0;
        t = 3;
        d = 1;
        sumad = 0;
        
        while (k<3) {
            while ((j<t) && (t<=5)) {
                d = d * A[i][j];
                System.out.println("D = "+d);
                i = i + 1;
                j = j + 1;
            }
            sumad = sumad + d;
            System.out.println("*** Suma D["+i+"] = "+sumad);
            d = 1;
            i = 0;
            k = k + 1;
            dc = dc + 1;
            j = dc;
            t = t + 1;
        }
        i = 0; j = 2; dc = 2; k = 0;
        t = 0; z = 1; sumai = 0;
        
        while (k<3) {
            while ((j>=t) && (t<3)) {
                z = z * A[i][j];
                i = i + 1;
                j = j - 1;
                System.out.println("Z = "+z);
            }
            sumai = sumai + z;
            System.out.println("*** Suma I ["+i+"] = "+sumai);
            z = 1;
            i = 0;
            k = k + 1;
            dc = dc + 1;
            j = dc;
            t = t + 1;
        }
        System.out.println("sumad = "+sumad+" - sumai = "+sumai);
        det = sumad - sumai;
        return det;
    }
}
